server/src/main/java/org/allaymc/server/network/processor/ingame/InventoryTransactionPacketProcessor.java

```diff
@@ -91,27 +91,25 @@ public void handleSync(Player player, InventoryTransactionPacket packet, long re
 
                         entity.setUsingItemInAir(false);
                         itemInHand.rightClickItemOnBlock(dimension, placeBlockPos, interactInfo);
-                        if (entity.isUsingItemOnBlock()) {
-                            if (itemInHand.useItemOnBlock(dimension, placeBlockPos, interactInfo)) {
-                                // Using item on the block successfully, no need to call BlockBehavior::onInteract()
-                                break;
-                            }
+                        if (itemInHand.useItemOnBlock(dimension, placeBlockPos, interactInfo)) {
```

Comment:
I don't like this solution very much, it's too hacky. I think I'll investigate this issue further